src/benchmarklib/synthetic_table_generator.hpp

```diff
@@ -0,0 +1,139 @@
+                                                          double skew_shape = 0.0) {
+    ColumnDataDistribution c{};
+    c.skew_location = skew_location;
+    c.skew_scale = skew_scale;
+    c.skew_shape = skew_shape;
+    c.distribution_type = DataDistributionType::NormalSkewed;
+    return c;
+  }
+
+  DataDistributionType distribution_type = DataDistributionType::Uniform;
+
+  int num_different_values = 1'000;
+
+  double pareto_scale = 1.0;
+  double pareto_shape = 1.0;
+
+  double skew_location = 0.0;
+  double skew_scale = 1.0;
+  double skew_shape = 0.0;
+
+  double min_value = 0.0;
+  double max_value = 1.0;
+};
+
+class SyntheticTableGenerator {
+  // Note: numa_distribute_chunks=true only affects generated tables that use DictionaryCompression,
+  // otherwise the chunks are most likely all placed on a single node. This might change in the future.
+  // See the discussion here https://github.com/hyrise/hyrise/pull/402
+ public:
+  // Simple table generation, mainly for simple tests
+  std::shared_ptr<Table> generate_table(const size_t num_columns, const size_t num_rows, const ChunkOffset chunk_size,
+                                        const SegmentEncodingSpec segment_encoding_spec = {EncodingType::Unencoded});
+
+  std::shared_ptr<Table> generate_table(const std::vector<ColumnDataDistribution>& column_data_distributions,
+                                        const std::vector<DataType>& column_data_types, const size_t num_rows,
+                                        const ChunkOffset chunk_size,
+                                        const std::vector<SegmentEncodingSpec>& segment_encoding_specs,
+                                        const std::optional<std::vector<std::string>> column_names = std::nullopt,
+                                        const UseMvcc use_mvcc = UseMvcc::No,
+                                        const bool numa_distribute_chunks = false);
+
+  // Base function that generates the actual data
+  std::shared_ptr<Table> generate_table(const std::vector<ColumnDataDistribution>& column_data_distributions,
+                                        const std::vector<DataType>& column_data_types, const size_t num_rows,
+                                        const ChunkOffset chunk_size,
+                                        const std::optional<std::vector<std::string>> column_names = std::nullopt,
+                                        const UseMvcc use_mvcc = UseMvcc::No,
+                                        const bool numa_distribute_chunks = false);
+
+  /**
+    * Function to cast an integer to the requested type.
+    *   - in case of long, the integer is simply casted
+    *   - in case of floating types, the integer slighlty modified
+    *     and casted to ensure a matissa that is not fully zero'd
+    *   - in case of strings, a 10 char string is created that has at least
```

Comment:
Does this make the point a little bit clearer?

```
    *   - in case of strings, a 10 char string is created that has at least `prefix_length` leading spaces to ensure
    *     that scans have to evaluate at least the first four chars. The reason is that very often strings are dates in
    *     in ERP systems and we assume that at least the year has to be read before a non-match can be determined.
    *     Randomized strings often lead to unrealistically fast string scans. An example would be randomized strings in
    *     which a typical linear scan only has to read the first char in order to disqualify a tuple. Strings in real
    *     world systems often share a common prefix (e.g., country code prefixes or dates starting with the year) where
    *     usually more chars need to be read. The observed effect was that operations on randomized strings were
    *     unexpectedly faster than seen with real-world data.
```

Reply:
Better, but I was actually asking for an example string generated by a certain input value

Reply:
https://gist.github.com/Bouncner/fe2cef4e3aaeea31612faab95999f3a1

Values 1 to 999.

Reply:
Add two examples to the comment?